File: Math/#07-Reverse Integer-Medium/ReverseInteger.c

#include<stdio.h>
int reverse(int x){
    int MAX_INT=pow(2,31)-1;
    int MIN_INT=pow(-2,31);
    if (x==0 || (x>-10 && x<10)){
        return x;
    }
    int sign;
    if(x>=0) sign=1; else sign=-1;
    long int xx=labs(x);
    long int rev=0;
    while(xx){
        rev=rev*10 + xx%10;
        xx=xx/10;
    }
    if(rev*sign>MAX_INT || rev*sign<MIN_INT){
        return 0;
    }
    return rev*sign;
    
}

int main(){
    printf("%d",reverse(13452));
}

File: Math/#07-Reverse Integer-Medium/ReverseInteger.py

def Solution( x: int) -> int:
        MAX_INT=(2**31)-1
        MIN_INT=-2**31
        if -10<x<10==0:
            return x
        sign = (x)//abs(x)
        rev = int(str(abs(x))[::-1])
        if(sign==-1 and rev*sign<MIN_INT):
            return 0
        elif(sign==1 and rev>MAX_INT):
            return 0
        else:
            return rev*sign
             
if __name__ == "__main__":
    print(Solution(456))
